pkg/kubectl/rolling_updater.go

```diff
@@ -208,9 +208,15 @@ func (r *RollingUpdater) Update(config *RollingUpdaterConfig) error {
 	if err != nil {
 		return err
 	}
-	// Further validation.
-	if maxUnavailable == 0 && maxSurge == 0 {
-		return fmt.Errorf("one of maxSurge or maxUnavailable must be specified")
+	// If the desired new scale is 0, then the max unavailable is necessarily
+	// the effective scale of the old RC regardless of the configuration
+	// (equivalent to 100% maxUnavailable).
+	if desired == 0 {
+		maxUnavailable = int(math.Max(float64(oldRc.Spec.Replicas), float64(oldRc.Status.Replicas)))
```

Comment:
That fixes the desired = 0 case.
original = 30. desired = 2 and maxUnavailable = 50% is still broken.

In your case, original=0, desired=10, maxUnavailable=50%, yes we will not scale down which is correct. There is no old pod to scale down :)

I hope that the tests will fail if we miss a case :)